DirectoryTree - Missing custom field support in typescript interface

lukasrandom opened the pull request:
Hey,
just tried to add custom field to DirectoryTree object in typescript inside a DirectoryTreeCallback. The following error occured:

### Error:
```
error TS7053: Element implicitly has an 'any' type because expression of type '"id"' can't be used to index type 'DirectoryTree'.
  Property 'id' does not exist on type 'DirectoryTree'.

34     item['id'] = createHash('sha1').update(path).digest('base64');
```

### Example adding custom id field:
```
import directoryTree, {
    DirectoryTree,
    DirectoryTreeCallback,
} from 'directory-tree';
import { createHash } from 'crypto';

const path = 'C:/Users/me/Desktop/a';
const callback: DirectoryTreeCallback = (item: DirectoryTree, path: string) => {
    item['id'] = createHash('sha1').update(path).digest('base64');
};
const dirTree: DirectoryTree = directoryTree(path, {}, callback, callback);
console.log(JSON.stringify(dirTree, null, 2));
```
### Fix:

Added `[key: string]: any` in to the `DirectoryTree `interface at` node_modules\directory-tree\index.d.ts`.


Already opened an issue here:
https://github.com/mihneadb/node-directory-tree/issues/103

Best
Lukas

mihneadb commented:
@lukasrandom hey, thanks for this! Do you think maybe it'd make more sense to group all of these into some sort of `extra` object? So at least they are still packed together? Happy to accept the PR otherwise.
Thanks!

lukasrandom commented:
@mihneadb 
Thanks to the feedback :)
New commit adds the "custom" field on the "DirectoryTree" interface to encapsulate additional extra information.

mihneadb commented:
@lukasrandom thanks! Looks good, with the mention that it appears the indenting was changed from spaces to tabs :S. Please leave it as it was. Oh and it might be worth adding 1-2 words about this new option in the Readme, if you find a place for them.
Thanks!

lukasrandom commented:
Fixed the formatting issue to stay to the original. Didn't noticed that prettier was changing the formatting.
Added a small readme entry.
Commit: 2e89215fc2801ea84d5c6a0ae84fc26fd58aed45
